### ISSUE TITLE
Toggling Light/Dark Theme has no effect on Card displayed on DialogHost

### ISSUE THREAD

**Issue by andybond006** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#571]
**Issue:**
When using the command `new PaletteHelper().SetLightDark(true)` the opened Card Dialog will not toggle its theme. The MainWindow (behind the Card Dialog) does toggle correctly in the background.

**Additional Testing:**
I also tested `new PaletteHelper().ReplacePrimaryColor(swatch)` and `new PaletteHelper().ReplaceAccentColor(swatch)` and both seem to work correctly.

**Repository:**
https://github.com/andybond006/MaterialDesignDialogColorDemo

**To reproduce issue:** 
1. Run the solution
2. Click the only button available to bring up the Card Dialog
3. Use the toggles to test issue

**Example Images:**

_Example Before:_
![before - lighttheme](https://cloud.githubusercontent.com/assets/10934345/21018565/5338a05c-bd3b-11e6-8350-4cac3b52076b.PNG)

_Example After:_
![after - darktheme](https://cloud.githubusercontent.com/assets/10934345/21018584/60d27062-bd3b-11e6-907d-1ba23d2ba04f.PNG)

**Comment by ButchersBoy** [comment]
Ah, when we spoke we only spoke about colour.  Dialogs are always light, by design.  At least currently :)

I will look at palette first.

**Comment by andybond006** [comment]
Okay, I understand if that's how it is by design. Sorry for the confusion.

I can see, however, that if I gave a user the option to "Toggle Light/Dark Theme" they may be as confused as I was, however, when some elements (Dialogs) remained Light Themed if they chose Dark.

Thoughts? If this was added as a possible feature would the darkening of the background be too close to the color of a Dark Themed Card?

**Comment by Awsmolak** [comment]
I was recently confused by this as well and found this old issue.

Now that "dark" themed apps have become more widespread, I can definitely say that the choice to keep the dialogs light seems to go against convention. I think it could be added as an option. But also, I'd argue that the dialogs should be dark by default. And I do realize that it's just an opinion, but the appearance of a white dialog when using a dark app is very visually jarring.

**Comment by Keboo** [comment]
This has actually been fixed in the latest version (3.1.0).
The changes needed to do it:
1. Change the DialogHost.DialogTheme property to "Inherit" so the dialog picks up the theme of the app.
2. Use the new [theming API](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/wiki/Advanced-Theming) to change from light/dark

Example of what I did in your app:
```
private void LightDark_ToggleBTN_Click(object sender, RoutedEventArgs e)
{
    var helper = new PaletteHelper();
    ITheme theme = helper.GetTheme();

    theme.SetBaseTheme((bool)LightDark_ToggleBTN.IsChecked ? Theme.Dark : Theme.Light);

    helper.SetTheme(theme);
}
```
